### ISSUE TITLE
rust version sdk needed.

### ISSUE THREAD

**Issue by leochan007** [ccxt/ccxt#24398]
### Preliminary Checks

- [X] I have already searched for [existing issues](https://github.com/ccxt/ccxt/issues) and confirmed that this issue is not a duplicate

### Is your feature request related to a problem? Please describe

want write rust crypto trading bot. but there is no rust version ccxt.

### Describe the solution you'd like

gen rust version sdk.

**Comment by carlosmiei** [comment]
Hello @leochan007,

Thanks we will take your suggestion into consideration.

The next language to be released is GO, but Rust might be an option for the next language after go.

**Comment by leochan007** [comment]
@carlosmiei thx man. u know now sol/rust is too hot for years. rust is also popular in web3.

**Comment by namn-grg** [comment]
I second this. @carlosmiei I just sent an email to info@ccxt . Anyone else interested I am considering implementing OS ourselves.
Reach out if interested :)

**Comment by Roh-codeur** [comment]
+1. Any update on this please? how can the community help? Found the below:

https://github.com/kizzx2/ccxt-rs

dont think this is maintained 

thanks guys

**Comment by 731315163** [comment]
yes we need rust

**Comment by 731315163** [comment]
https://github.com/ccxt/ast-transpiler/pull/28#issue-3040144558

**Comment by Aviksaikat** [comment]
I agree with this. Considering the demand of `rust` in the web3 space we need a section for rust as well.

**Comment by dima-dmytruk23** [comment]
+1

**Comment by TaiShanQ** [comment]
+1

**Comment by lingcoder** [comment]
+1

**Comment by Aviksaikat** [comment]
Happy to work on the rust version

**Comment by ligl** [comment]
我们需要一个Rust版本

**Comment by Aviksaikat** [comment]
Since this issue is not getting enough attention from the devs, I've started making a rust version here https://github.com/Ankvik-Tech-Labs/ccxt-rs 

Can't promise I can support every exchange. I'm working whenever I get time. You can trust this as we've created the patch to support `GMX` in `CCXT`

**Comment by ttodua** [comment]
CCXT Rust is already in progress, you can watch this PR for updates: https://github.com/ccxt/ccxt/pull/28627

**Comment by kroitor** [comment]
CCXT now officially speaks Rust 🦀

Starting from **v4.5.78**, the Rust version is publicly released. Three crates are live on crates.io:

- [`ccxt`](https://crates.io/crates/ccxt) — REST exchanges
- [`ccxt-pro`](https://crates.io/crates/ccxt-pro) — WebSocket (`watch_*`) exchanges
- [`ccxt-prediction`](https://crates.io/crates/ccxt-prediction) — prediction markets

Requirements: Rust stable (edition 2021) and a [tokio](https://tokio.rs) runtime.

```rust
use ccxt::{Binance, Config, Params};

#[tokio::main]
async fn main() -> Result<(), ccxt::ExchangeError> {
    let mut exchange = Binance::with_config(
        Config::new()
            .api_key("your api key")
            .secret("your secret"),
    );

    let order = exchange
        .create_order("ETH/USDT", "market", "buy", 0.1)
        .await?;
    println!("{:?} {:?}", order.id, order.status);

    Ok(())
}
```

More examples: https://github.com/ccxt/ccxt/tree/master/examples/rs

📌 Announcement: https://github.com/ccxt/ccxt/issues/30309
🐦 https://x.com/ccxt_official/status/2097034527356993873

Thanks to everyone who pushed for this over the years — the demand in this thread is a big part of why it happened. Special thanks to @Aviksaikat for building [ccxt-rs](https://github.com/Ankvik-Tech-Labs/ccxt-rs) while this was in the works. 🙏

If you run into anything with the Rust version, please open a new issue — feedback from early adopters is exactly what we need right now.

<img width="3200" height="1800" alt="CCXT now in Rust — crates ccxt, ccxt-pro, ccxt-prediction live on crates.io starting v4.5.78" src="https://github.com/user-attachments/assets/9014b0e3-9b66-4eb8-9b07-d5cd393474f9" />
